File: sorts/Bubble sort/Bubble sort.cpp
#include <iostream>

using namespace std;



///
// Sort function
///
void BubleSort(int *arr, int size) {
	for (int i = 0; i < size - 1; i++) {
		for (int j = 0; j < size - i - 1; j++) {
			if (arr[j] > arr[j + 1]) {
				swap(arr[j], arr[j + 1]);
			}
		}
	}
}
///
// End of sort function
///
int main()
{
	///
	// Exemple of usage
	///
	int *arr;
	int size;
	cout << "n = ";
	cin >> size;
	
	arr = new int[size];


	///
	// Read array
	///
	for (int i = 0; i < size; i++) {
		cout << "arr[" << i << "] = ";
		cin >> arr[i];
	}
	
	

	///
	// Calling a sort function
	///
	BubleSort(arr, size);


	///
	// Output result
	///
	for (int i = 0; i < size; i++) {
		cout << arr[i] << " ";
	}
	cout << endl;
	delete[] arr;
	return 0;
	///
	//  End of exemple of usage
	///
}

File: Sorts/Bubble sort/Bubble sort.py
###
## Start of sort function
###
def BubleSort(arr):
    i = 0
    while i < len(arr) - 1:
        j = 0
        while j < len(arr) - i - 1:
            if arr[j] > arr[j + 1]:
                arr[j], arr[j + 1] = arr[j + 1], arr[j]
            j += 1
        i += 1
    return arr
###
## End of sort function
###


###
## Exemple of usage
###


arr = []
size = int(input("n = "))

###
## Read array
###
i = 0
while i < size:
    arr.append(int(input("arr[" + str(i) + "] = ")))
    i += 1

###
##  Calling a sort function
###
arr = BubleSort(arr)



###
## Output result
###

print(arr)

###
## End of exemple of usage
###
